src/xml.ts

```diff
@@ -115,10 +115,10 @@ function getAllTests(task: Task): Test[] {
 }
 
 function getDurationAttribute(test: Test): string {
-    const duration = test.result?.duration;
+    const duration = test.result?.duration ?? test.file?.result?.duration;
```

Comment:
`test.file?.result` - I'm not sure about this one. What exactly is the `test.file.result`? A `describe` block?

I think the `jest-sonar-reporter` is using only `test.result`:
- https://github.com/3dmind/jest-sonar-reporter/blob/develop/lib/xml/testCase.js#L10
- https://github.com/facebook/jest/blob/main/packages/jest-types/src/TestResult.ts#L20

Reply:
Looks like it's the sum of all other tests. Duration on line 5 is the sum of lines 3, 9 and 13. 

```xml
 1 | <testExecutions version="1">
 2 |   <file path="test/fixtures/animals.test.ts">
 3 |     <testCase name="animals - dogs say woof" duration="2" />
 4 | 
 5 |     <testCase name="animals - figure out what rabbits say" duration="7">
 6 |       <skipped message="figure out what rabbits say" />
 7 |     </testCase>
 8 | 
 9 |     <testCase name="animals - flying ones - cat can fly" duration="4">
10 |       <failure message="expected false to be true // Object.is equality">...</failure>
11 |     </testCase>
12 | 
13 |     <testCase name="animals - flying ones - bird can fly" duration="1" />
14 |   </file>
15 | </testExecutions>
```

I think correct way is to set skipped tests with `duration="0"`:
```suggestion
    const duration = test.result?.duration;
```